pkg/cloudprovider/metrics/cloudprovider.go

```diff
@@ -0,0 +1,143 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"context"
+	"time"
+
+	"github.com/aws/karpenter/pkg/apis/provisioning/v1alpha5"
+	"github.com/aws/karpenter/pkg/cloudprovider"
+	"github.com/aws/karpenter/pkg/metrics"
+	"github.com/aws/karpenter/pkg/utils/injection"
+	"github.com/prometheus/client_golang/prometheus"
+	v1 "k8s.io/api/core/v1"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/logging"
+	crmetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+const (
+	metricLabelComponent = "component"
+	metricLabelMethod    = "method"
+	metricLabelProvider  = "provider"
+	metricLabelResult    = "result"
+)
+
+var methodDurationHistogramVec = prometheus.NewHistogramVec(
+	prometheus.HistogramOpts{
+		Namespace: metrics.Namespace,
+		Subsystem: "cloudprovider",
+		Name:      "duration_seconds",
+		Help:      "Duration of cloud provider method calls.",
+	},
+	[]string{
+		metricLabelComponent,
+		metricLabelMethod,
+		metricLabelProvider,
+		metricLabelResult,
```

Comment:
I'm not sure we want to track the result of these calls. I'd delegate error tracking elsewhere. This will allow you to simplify and use `metrics.Measure()`